src/js/components/Layer/LayerContainer.js

```diff
@@ -44,6 +44,25 @@ const LayerContainer = forwardRef(
     const containerRef = useRef();
     const layerRef = useRef();
 
+    useEffect(() => {
+      const handleClick = event => {
+        if (containerRef.current.contains(event.target)) {
```

Comment:
This approach won't work for a Layer that uses a Drop. For instance, if the Layer has a Select control, when the use opens the Select drop down and clicks on it, this will close the Layer. You can see this in the Layer -> Center story. I think this is why https://v2.grommet.io/layer#onClickOutside documents that it is only available for modal Layers.

Reply:
I changed it to Only work on modal layers and if you click on the overlay